InterfaceFortran/fortester.f90

```diff
@@ -27,184 +27,1471 @@
+     check_points( points(2,:), (/12345.d0/), (/-0.75d0,-0.25d0,0.25d0,0.75d0/) ) .or. &
+     check_points( points(3,:), (/12345.d0/), (/12345.d0/) ) ) then
+  write(*,*) "Mismatch in tsgSetSurplusRefinement: limits refine using existing limits"
+  stop 1
+endif
+call tsgSetLocalSurplusRefinement(gridID,1.d-8,tsg_classic,1,(/2,2,3/))
+if ( tsgGetNumNeeded(gridID) .eq. 0 ) then
+  write(*,*) "Mismatch in tsgSetLocalSurplusRefinement: did not refine on second pass"
+  stop 1
+endif
+if ( check_points( points(1,:), (/0.5d0,-0.5d0/), (/-0.75d0,-0.25d0,0.25d0,0.75d0/) ) .or. &
+     check_points( points(2,:), (/12345.d0/), (/-0.75d0,-0.25d0,0.25d0,0.75d0/) ) ) then
+  write(*,*) "Mismatch in tsgSetSurplusRefinement: limits refine using new limits"
+  stop 1
+endif
+deallocate(points, double_2d_a)
+
+
+
+
+
+
+
+
+!=======================================================================
+!       tsgClearRefinement()
+!=======================================================================
+call tsgMakeLocalPolynomialGrid(gridID, 3, 1, 4, 2, tsg_localp, levelLimits=(/300,300,300/))
+points => tsgGetPoints(gridID);  i_b = tsgGetNumPoints(gridID)
+allocate( double_2d_a(1,i_b) )
+double_2d_a(1,:) = exp( -points(1,:)**2 - 0.5*points(2,:)**2 - 2.d0*points(3,:)**2 )
+call tsgLoadNeededPoints(gridID, double_2d_a)
+if ( tsgGetNumNeeded(gridID) > 0 ) then
+  write(*,*) "Mismatch in cancel refine: did not load values"
+  stop 1
+endif
+call tsgSetLocalSurplusRefinement(gridID,1.d-4,tsg_directional)
+if ( tsgGetNumNeeded(gridID) .eq. 0 ) then
+  write(*,*) "Mismatch in cancel refine: did not set refinement at output -1"
+  stop 1
+endif
+call tsgClearRefinement(gridID)
+if ( tsgGetNumNeeded(gridID) > 0 ) then
+  write(*,*) "Mismatch in cancel refine: did not cancel the refinement"
+  stop 1
+endif
+deallocate(points, double_2d_a)
+
+
+
+
+
+
+
+
+!=======================================================================
+!       tsgMergeRefinement()
+!=======================================================================
+call tsgMakeGlobalGrid(gridID,    2, 1, 4, tsg_level, tsg_fejer2)
+call tsgMakeGlobalGrid(gridID_II, 2, 1, 4, tsg_level, tsg_fejer2)
+
+points => tsgGetPoints(gridID);  i_b = tsgGetNumPoints(gridID)
+allocate( double_2d_a(1,i_b) )
+double_2d_a(1,:) = exp( -points(1,:)**2 - points(2,:)**2 )
+call tsgLoadNeededPoints(gridID,    double_2d_a)
+call tsgLoadNeededPoints(gridID_II, double_2d_a)
+call tsgSetAnisotropicRefinement( gridID,    tsg_iptotal, 10, 1 )
+call tsgSetAnisotropicRefinement( gridID_II, tsg_iptotal, 10, 1 )
+deallocate(points, double_2d_a)
+
+points => tsgGetNeededPoints(gridID);  i_b = tsgGetNumNeeded(gridID)
+allocate( double_2d_a(1,i_b) )
+double_2d_a(1,:) = exp( -points(1,:)**2 - points(2,:)**2 )
+call tsgLoadNeededPoints(gridID, double_2d_a)
+call tsgMergeRefinement(gridID_II)
+deallocate(points, double_2d_a)
+
+i_a = 20
+allocate( pointsb(2,i_a) );
```

Comment:
* here you're allocating `pointsb` only to then have it immediately reassigned in line 1349. Get rid of this line.